### PR TITLE
[CPP20] Fix compilation issue in cmssw/DataFormats/HcalDigi/test/test_hcal_digi.cu

### LINKED ISSUE #42624: [CPP20] Error: function "std::__detail::__variant::_Variadic_union<_First, _Rest...>::~_Variadic_union()" is deleted
During compilation of DataFormats/HcalDigi package, the following errors are emitted:

```

### PR DESCRIPTION
**Pull request by iarspider**
#### PR description:

Fixes https://github.com/cms-sw/cmssw/issues/42624 as suggested by @makortel in https://github.com/cms-sw/cmssw/issues/42624#issuecomment-1688215545

#### PR validation:

Bot tests.

### CODE REVIEW

**Comment by iarspider** [comment]
test parameters:
- addpkg = DataFormats/HcalDigi,DataFormats/Common

**Comment by iarspider** [comment]
please test for CMSSW_13_3_CPP20_X

**Comment by iarspider** [comment]
please test for CMSSW_13_3_CPP20_X

File: DataFormats/FTLDigi/src/classes_def.xml [hunk 1]
```diff
@@ -4,24 +4,20 @@
    <class name="FTLDataFrame"/>
    <class name="std::vector<FTLDataFrame>"/>
    <class name="FTLDigiCollection"/>
-   <class name="edm::Wrapper<FTLDigiCollection>" />
```
Comment (makortel) [inline DataFormats/FTLDigi/src/classes_def.xml:7]: Did you verify that only the `Wrapper<T>` is not used? (I don't disagree, on a quick look the entire `FTLDigiCollection` seem unused, in which case I'd proceed to remove them all, but I'd also prefer such cleanup to be done in a separate PR)

File: DataFormats/Common/interface/SortedCollection.h [hunk 2]
```diff
@@ -28,7 +28,8 @@ unreliable if such duplicate entries are made.
 #include "DataFormats/Common/interface/CMS_CLASS_VERSION.h"
 #include "DataFormats/Common/interface/fillPtrVector.h"
 #include "DataFormats/Common/interface/FillView.h"
-#include "DataFormats/Common/interface/Ref.h"
+// #include "DataFormats/Common/interface/Ref.h"
```
Comment (makortel) [inline DataFormats/Common/interface/SortedCollection.h:31]: No need to leave commented out line
```suggestion
```

File: DataFormats/FTLDigi/src/classes_def.xml (hunk 1 above)
Comment (iarspider) [reply DataFormats/FTLDigi/src/classes_def.xml:7]: @makortel sorry if I did something wrong: I just cleaned the entries that were causing the build to fail. I am running a local build now to check if there are other unused entries.

Comment (makortel) [reply DataFormats/FTLDigi/src/classes_def.xml:7]: The error from `rootcling` is indeed a bit strange. But removing the `Wrapper<T>` from the `classes_def.xml` is unlikely to be the right fix (`rootcling` doesn't know if the `Wrapper<T>` is really needed or not).

Just guessing, would the error disappear if the `DataFormats/Common/interface/Ref.h` would be included in the `classes.h`?

Comment (iarspider) [reply DataFormats/FTLDigi/src/classes_def.xml]: Yes, adding that does work for local build.

**Comment by iarspider** [comment]
please test for CMSSW_13_3_CPP20_X

File: DataFormats/FTLDigi/src/classes_def.xml (hunk 1 above)
Comment (makortel) [reply DataFormats/FTLDigi/src/classes_def.xml]: Thanks, interesting discovery. I wonder how including `Ref.h` plays a role here.

**Comment by makortel** [comment]
@cmsbuild, please test

Let's run normal tests as well

**Comment by jfernan2** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

Technical PR.

**Comment by antoniovilela** [comment]
+1
